### PR TITLE
DropdownButton RTL

### PR DESCRIPTION
**Pull request by Hixie**
This fixes DropdownButtons to align their popups correctly in RTL as well.

Also while I was there I fixed the issue with text scale factor in the gallery.

cc @gspencergoog

### CODE REVIEW

File: packages/flutter/test/material/dropdown_test.dart [hunk 1]
```diff
@@ -252,37 +284,50 @@ void main() {
...
-
-    await tester.pumpWidget(build());
-    final RenderBox buttonBox = tester.renderObject(find.byKey(buttonKey));
-    assert(buttonBox.attached);
-    final Offset buttonOriginBeforeTap = buttonBox.localToGlobal(Offset.zero);
-
-    await tester.tap(find.text('two'));
-    await tester.pump();
-    await tester.pump(const Duration(seconds: 1)); // finish the menu animation
+    for (TextDirection textDirection in TextDirection.values) {
```
Comment (abarth) [inline packages/flutter/test/material/dropdown_test.dart]: I've been using separate tests for the directions to make the failures more obvious.  When this test fails, the output won't tell you which of the directions failed (or if they both fail).  Consider extracting the body of the test into a function and having an RTL and LTR test call it.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
That was a lot less scary than I imagined it.

File: packages/flutter/test/material/dropdown_test.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/test/material/dropdown_test.dart]: Good point. I inverted the for loop and the test declaration:

```dart
  for (TextDirection textDirection in TextDirection.values) {
    testWidgets('Dropdown button aligns selected menu item ($textDirection)', (WidgetTester tester) async {
```

**Comment by Hixie** [comment]
Thanks for the review. Will land on green.

**Pull request merged**